### PR TITLE
Utilize CreateOrUpdate for all Infinispan CR's updates in test

### PR DESCRIPTION
**Pull request by dmvolod**

### CODE REVIEW

**Review by rigazilla — APPROVED** [review]

File: test/e2e/utils/kubernetes.go [hunk 1]
```diff
@@ -236,10 +226,26 @@ func (k TestKubernetes) GracefulRestartInfinispan(infinispan *ispnv1.Infinispan,
...
 		return true, nil
 	})
-
 	ExpectNoError(err)
 }
 
+func (k TestKubernetes) UpdateInfinispan(ispn *ispnv1.Infinispan, update func()) error {
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), k.Kubernetes.Client, ispn, func() error {
+		if ispn.CreationTimestamp.IsZero() {
+			return errors.NewNotFound(ispnv1.Resource("infinispan.org"), ispn.Name)
```
Comment (ryanemerson) [inline test/e2e/utils/kubernetes.go:235]: If we thrown a `NotFound` err on a create event, why not just use `client.Update` directly?

Comment (dmvolod) [reply test/e2e/utils/kubernetes.go:235]: CreateOrUpdate do much more than update, it refresh data from CR, compare and Update if needed. We can create or own solution, but I'm prefer to use a library functions.

Comment (ryanemerson) [reply test/e2e/utils/kubernetes.go:235]: Right makes sense. I thought I was probably missing something.

**Comment by ryanemerson** [comment]
Thanks @dmvolod

**Pull request merged**